go/signature/signature_key_templates_test.go

```diff
@@ -19,65 +19,170 @@ import (
+			hashType:    commonpb.HashType_SHA512,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_TINK,
+		},
+		{
+			tcName:      "P-256 with SHA256, DER format and RAW output prefix",
+			typeURL:     testutil.ECDSASignerTypeURL,
+			sigTemplate: signature.ECDSAP256KeyWithoutPrefixTemplate(),
+			curveType:   commonpb.EllipticCurveType_NIST_P256,
+			hashType:    commonpb.HashType_SHA256,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_RAW,
+		},
+		{
+			tcName:      "P-384 with SHA512, DER format and RAW output prefix",
+			typeURL:     testutil.ECDSASignerTypeURL,
+			sigTemplate: signature.ECDSAP384KeyWithoutPrefixTemplate(),
+			curveType:   commonpb.EllipticCurveType_NIST_P384,
+			hashType:    commonpb.HashType_SHA512,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_RAW,
+		},
+		{
+			tcName:      "P-521 with SHA512, DER format and RAW output prefix",
+			typeURL:     testutil.ECDSASignerTypeURL,
+			sigTemplate: signature.ECDSAP521KeyWithoutPrefixTemplate(),
+			curveType:   commonpb.EllipticCurveType_NIST_P521,
+			hashType:    commonpb.HashType_SHA512,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_RAW,
+		},
 	}
-	// ECDSA P-384
-	template = signature.ECDSAP384KeyTemplate()
-	err = checkECDSAKeyTemplate(template,
-		commonpb.HashType_SHA512,
-		commonpb.EllipticCurveType_NIST_P384,
-		ecdsapb.EcdsaSignatureEncoding_DER)
-	if err != nil {
-		t.Errorf("invalid ECDSA P-384 key template: %s", err)
+
+	for _, tt := range flagTests {
+		t.Run("test ECDSA - "+tt.tcName, func(t *testing.T) {
+			err := checkECDSAKeyTemplate(tt.sigTemplate,
+				tt.typeURL,
+				tt.hashType,
+				tt.curveType,
+				tt.sigEncoding,
+				tt.prefixType)
+			if err != nil {
+				t.Errorf("failed %s: %s", tt.tcName, err)
+			}
+		})
 	}
-	// ECDSA P-521
-	template = signature.ECDSAP521KeyTemplate()
-	err = checkECDSAKeyTemplate(template,
-		commonpb.HashType_SHA512,
-		commonpb.EllipticCurveType_NIST_P521,
-		ecdsapb.EcdsaSignatureEncoding_DER)
-	if err != nil {
-		t.Errorf("invalid ECDSA P-521 key template: %s", err)
+}
+
+func TestEDDSAKeyTemplates(t *testing.T) {
+	var flagTests = []struct {
+		tcName      string
+		typeURL     string
+		sigTemplate *tinkpb.KeyTemplate
+		prefixType  tinkpb.OutputPrefixType
+	}{
+		{
+			tcName:      "ED25519 with TINK output prefix",
+			typeURL:     testutil.ED25519SignerTypeURL,
+			sigTemplate: signature.ED25519KeyTemplate(),
+			prefixType:  tinkpb.OutputPrefixType_TINK,
+		},
+		{
+			tcName:      "ED25519 with RAW output prefix",
+			typeURL:     testutil.ED25519SignerTypeURL,
+			sigTemplate: signature.ED25519KeyWithoutPrefixTemplate(),
+			prefixType:  tinkpb.OutputPrefixType_RAW,
+		},
+	}
+
+	for _, tt := range flagTests {
+		t.Run("Test EDDSA - "+tt.tcName, func(t *testing.T) {
+			err := checkDSAKeyTemplate(tt.sigTemplate,
```

Comment:
DSA is actually a name of another digital signature algorithm, so this is while confusing.

I suggest renaming is to checkKeyTypeAndOutputPrefix

Reply:
done